### ISSUE TITLE
Home assistant keeps crashing - Exception in callback BermudaDataUpdateCoordinator._refresh_scanners.<locals>.async_call_update_entry()

### ISSUE THREAD

**Issue by Ill-tableflip-U** [agittins/bermuda#351]
Bermuda keeps crashing home assistant entirely - I have to go into an SSH session and restart home assistant to fix it. This appears to be happening randomly when configuring it. How do I know it's bermuda? It is spamming the same bermuda error multiple times a second, throttling the entire home assistant installation. \:
[homeassistantlogs.txt](https://github.com/user-attachments/files/17606779/homeassistantlogs.txt)
This is a brand new bermuda installation, it worked fine for the first day and then just started doing this.
[config_entry-bermuda-01JBP9AE0GZAG40Z05EXGN37KD.json](https://github.com/user-attachments/files/17606787/config_entry-bermuda-01JBP9AE0GZAG40Z05EXGN37KD.json)
Also, I have been having problems with my esp32 boards not being used to scan for bluetooth at times. Don't know if this could be related?
![image](https://github.com/user-attachments/assets/45db394c-8e40-4c64-ada7-83000e74cda5)

**Comment by Lash-L** [comment]
That's actually a pretty crazy error.

Shouldn't be possible. Essentially, Home Assistant keeps track of every instance of every integration setup through a config entry id.

My only thought is that some other error is causing the config entry to be unloaded - but there is a pending task and then that task runs and looks for a config flow that is no longer up and running.

I would first try a uninstall and reinstall of the integration.

If that doesn't work, we probably should switch over to runtime_data instead of constantly updating the config flow.

**Comment by agittins** [comment]
Yeah, HA does things with config entries which I find confusing (which is almost certainly simply me not understanding the flow). Bermuda effectively reloads itself when it notices the scanner setup changing - but when it does it tends to trigger *multiple* reloads, and I think that on some systems this might be leading to an infinite loop of reloads.

I'm adding a check to ensure that the scanner config *is* actually different, and if not it skips that reload request. Because I haven't been able to replicate the issue I'm not 100% sure that it will solve this, but it should at least help.

I'll add another note when this change is available in a beta release, and it will be in the next full release.

@Lash-L moving to runtime_data is probably a good idea. I don't want to do it in the coming release as I need to get it out before 2024.11 drops, but I think it's worth doing since it's a perfect fit with what we are using the config options for currently. It'll be a fairly wide-ranging changeset though, and I think I'd like to revisit how "options" are handled generally - instead of them being duplicated through all the objects as they are currently.

**Comment by Ill-tableflip-U** [comment]
I completely removed bermuda from home assistant, and redownloaded it again using HACS, and I have been unable to replicate the issue again but I did experience this problem at least 3 times when it was installed before.

**Comment by agittins** [comment]
I have fixed a number of issues that could potentially cause race conditions on different systems. These are now released in v0.7.0.

I'll close this as "fixed", even though we haven't replicated it, since we have some likely fixes in the system now and also your system seems to be working OK - plus I have added a specific check for the particular error you experienced. Fingers crossed!

If you have any other issues come up feel free to raise another issue, or to re-open this one if the problem appears to be the same.
